script.js

```diff
@@ -0,0 +1,269 @@
+  const buttonsNewTask = document.createElement('div');
+  const imageDelete = document.createElement('img');
+  const imageEdit = document.createElement('img');
+  const buttonDelete = document.createElement('button');
+  const buttonEdit = document.createElement('button');
+  const oldText = document.createElement('p');
+  const checkBox = document.createElement('input');
+
+  checkBox.type = 'checkBox';
+  buttonEdit.id = `task__button_editing-${id}`;
+  buttonDelete.id = `task__button_deleting-${id}`;
+  oldText.id = `task__text-${id}`;
+  imageDelete.alt = 'Удалить';
+  imageEdit.alt = 'Добавить';
+  imageDelete.src = './images/delete.svg';
+  imageEdit.src = './images/edit.svg';
+  oldText.innerText = text;
+  newTask.className = 'header__task';
+  newTask.id = `task-${id}`;
+
+  buttonDelete.onclick = () => deleteTask(id);
+  buttonEdit.onclick = () => editTask(id, text);
+  checkBox.onclick = () => changeCheckBoxTask(id, isCheck);
+
+  buttonDelete.append(imageDelete);
+  buttonEdit.append(imageEdit);
+  buttonsNewTask.append(buttonEdit, buttonDelete);
+  newTask.append(checkBox, oldText, buttonsNewTask);
+  task.replaceWith(newTask);
+}
+
+const changeTextTask = async (id) => {
+  try {
+    const input = document.getElementById(`task__input-${id}`);  
+    if (input.value === null) {
+      return;
+    }
+    if (input.value === '') {
+      showError('Поле пустое');
+    }
+    const result = await fetch(`${link}/tasks/${id}/text`, {
+      method: 'PATCH',
+      headers: fetchHeaders,
+      body: JSON.stringify({
+        text: input.value
+      })
+    });
+    const data = await result.json();
+
+    allTasks.forEach(item => {
+      if (item._id === data._id) {
+        item.text = data.text;
+      }
+    });
+    render();
+  } catch (error) {
+    showError('Ошибка изменения задачи');
+  }
+};
+
+const editTask = (id, text, isCheck) => {
+  const task = document.getElementById(`task-${id}`);
+  const newTask = document.createElement('div');
+  const buttonsNewTask = document.createElement('div');
+  const oldText = document.getElementById(`task__text-${id}`);
+
+  const buttonChange = document.createElement('button');
+  const buttonCancel = document.createElement('button');
+  const imageChange = document.createElement('img');
+  const imageCancel = document.createElement('img');
+  const newText = document.createElement('input');
+
+  buttonChange.id = `task__button_change-${id}`;
+  buttonCancel.id = `task__button_cancel-${id}`;
+  newText.id = `task__input-${id}`;
+  newText.className = 'task__input_text';
+  newText.value = text;
+  newTask.className = 'header__task';
+  newTask.id = `task-${id}`;
+  imageCancel.alt = 'Отменить';
+  imageCancel.src = './images/cancel.png';
+  imageChange.alt = 'Изменить';
+  imageChange.src = './images/apply.png';
+  oldText.id = id;
+
+  buttonChange.onclick = () => changeTextTask(id);
+  buttonCancel.onclick = () => cancelTextTask(id, text, isCheck);

```

Comment:
where is isCheck from again? You also only pass two parameters above, not three.